python/tank/bootstrap/resolver.py

```diff
@@ -294,9 +295,60 @@ def _get_pipeline_configurations_for_project(self, pipeline_config_name, current
             "The following pipeline configurations were found: %s" % pprint.pformat(pipeline_configs)
         )
 
+        pcs = []
+
         for pc in pipeline_configs:
             path = ShotgunPath.from_shotgun_dict(pc)
 
+            # We'll need to provide a descriptor object for the config if
+            # possible. Note that it's possible that we'll be returning a
+            # None for the config descriptor. It's up to other filtering
+            # operations to remove those if desired.
+            try:
+                current_os_path = path.current_os
+            except ValueError:
+                current_os_path = None
+
+            uri = pc["descriptor"] or pc["sg_descriptor"]
+
+            if self._is_classic_pc(pc) or uri is None:
+                # If this is a classic-style config, we only concern ourselves with
+                # "*_path" fields and, specifically, the one that matches the current
+                # operating system. Likewise, if it's a zero-config setup, but without
+                # a descriptor uri set, we'll fall back on the path fields.
+                if current_os_path is None:
+                    log.debug("Config isn't setup for %s: %s", sys.path, pc)
+                    cfg_descriptor = None
+                else:
+                    try:
+                        cfg_descriptor = create_descriptor(
```

Comment:
Probably don't need to, given the amount of checking I'm doing before calling the create function.